app/frontend/src/features/contributorForm/ContributorForm.tsx

```diff
@@ -0,0 +1,367 @@
+import CircularProgress from "components/CircularProgress";
+import TextField from "components/TextField";
+import { useAuthContext } from "features/auth/AuthProvider";
+import { GetContributorFormInfoRes } from "pb/account_pb";
+import React, { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useMutation, useQuery } from "react-query";
+import { Link } from "react-router-dom";
+import { signupRoute } from "routes";
+import { service } from "service";
+
+import {
+  AGE,
+  ALREADY_FILLED_IN,
+  CONTRIBUTE_ARIA_LABEL,
+  CONTRIBUTE_LABEL,
+  CONTRIBUTE_OPTIONS,
+  CONTRIBUTE_REQUIRED,
+  EMAIL,
+  EMAIL_REQUIRED,
+  EXPERIENCE_HELPER,
+  EXPERIENCE_LABEL,
+  EXPERTISE_HELPER,
+  EXPERTISE_LABEL,
+  FEATURES_HELPER,
+  FEATURES_LABEL,
+  FILL_IN_AGAIN,
+  FILL_IN_THE_FORM,
+  GENDER,
+  GENDER_ARIA_LABEL,
+  GENDER_OPTIONS,
+  IDEAS_HELPER,
+  IDEAS_LABEL,
+  LOCATION_HELPER,
+  LOCATION_LABEL,
+  NAME,
+  NAME_REQUIRED,
+  PLEASE_SIGN_UP,
+  QUESTIONS_OPTIONAL,
+  SIGN_UP,
+  SUBMIT,
+  SUCCESS_MSG,
+  YOU_CAN_ALSO,
+} from "./constants";
+
+type ContributorInputs = {
+  username?: string;
+  name: string;
+  email: string;
+  contribute: string;
+  ideas: string;
+  features: string;
+  age: string;
+  gender: string;
+  location: string;
+  experience: string;
+  develop: string;
+  expertise: string;
+};
+
+type FormResponse = {
+  success: boolean;
+};
+
+const useStyles = makeStyles((theme) => ({
+  genderRadio: {
+    display: "flex",
+    flexDirection: "row",
+  },
+}));
+
+export default function ContributorForm() {
+  const classes = useStyles();
+  const { authState } = useAuthContext();
+
+  const [alreadyFilled, setAlreadyFilled] = useState(false);
+
+  const {
+    control,
+    register,
+    handleSubmit,
+    setValue,
+    errors,
+  } = useForm<ContributorInputs>({
+    defaultValues: {},
+    mode: "onBlur",
+    shouldUnregister: false,
+  });
+
+  const { isLoading: queryLoading, error: queryError } = useQuery<
+    GetContributorFormInfoRes.AsObject,
+    Error
+  >("contributorFormInfo", service.account.getContributorFormInfo, {
+    enabled: authState.authenticated,
+    onSuccess: (data) => {
+      setValue("username", data.username);
+      setValue("name", data.name);
+      setValue("email", data.email);
+      setValue("age", data.age);
+      setValue("gender", data.gender);
+      setValue("location", data.location);
+      setAlreadyFilled(data.filledContributorForm);
+    },
+  });
+
+  const postForm = async (data: ContributorInputs) => {
+    const response = await fetch(
+      "https://ja4o9uz9u3.execute-api.us-east-1.amazonaws.com/form_handler",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json;charset=UTF-8",
+        },
+        mode: "cors",
+        method: "POST",
+        body: JSON.stringify(data),
+      }
+    );
+    return (await response.json()) as FormResponse;
```

Comment:
```suggestion
    return response.json();
```
Unnecessary `await` if you are already returning a Promise - the caller (React Query) will await it correctly.